lwt/happy_eyeballs_lwt.ml

```diff
@@ -10,16 +10,24 @@ module Log = (val Logs.src_log src : Logs.LOG)
 
 let now = Mtime_clock.elapsed_ns
 
+type getaddrinfo = {
+  getaddrinfo : 'response 'a.
+    'response Dns.Rr_map.key -> 'a Domain_name.t -> ('response, [ `Msg of string ]) result Lwt.t
+} [@@unboxed]
+
 type t = {
   mutable waiters : ((Ipaddr.t * int) * Lwt_unix.file_descr, [ `Msg of string ]) result Lwt.u Happy_eyeballs.Waiter_map.t ;
   mutable cancel_connecting : (int * unit Lwt.u) list Happy_eyeballs.Waiter_map.t;
   mutable he : Happy_eyeballs.t ;
-  dns : Dns_client_lwt.t ;
   timer_interval : float ;
   timer_condition : unit Lwt_condition.t ;
   counter : int ;
+  mutable getaddrinfo : getaddrinfo ;
```

Comment:
maybe name it `resolver` or `dns`? Since the `t.getaddrinfo.getaddrinfo` looks weird.